### ISSUE TITLE
Nothing works

### ISSUE THREAD

**Issue by msmakhlouf** [vertexclique/tokamak#58]
After installing tokamak I try to build or run any of the commands nothing happens.

**Comment by dizergin** [comment]
confirm -win10 x64, clean atom installation.

**Comment by vertexclique** [comment]
@msmakhlouf can you give more detailed info about your current system?

**Comment by msmakhlouf** [comment]
Linux Mint 18
X64 
clean system install / atom install

**Comment by dizergin** [comment]
asus u32v, win10 x64 anniversary edition, clean (on the clean diskplace, with no 3-rd party packages), Atom editor v 1.98) , rust v1.10 was set with rust up....

**Comment by vertexclique** [comment]
@dizergin when you try to open a project with containing `tokamak.toml` how it behaves...

**Comment by dizergin** [comment]
hm... i CANNOT create a project with the such file

**Comment by vertexclique** [comment]
```
cargo new blabla
cd blabla
ECHO >> tokamak.toml
atom .
```
That is it actually. You can create such a file...

**Comment by dizergin** [comment]
of course i can... but i SAW no reason for... just give me a couple of minutes...

**Comment by msmakhlouf** [comment]
I tried that but it did not work

**Comment by dizergin** [comment]
update tokamak to 0.31-nothing changes
REINSTALL tokamak - AFTER restarting and opening of a rust project folder -got messages of the dependencies installations... After it got bunch of errors...

**Comment by mrmcq2u** [comment]
Have same issue with rustup on fedora 24.

**Comment by mrmcq2u** [comment]
@vertexclique - So when I launched an existing cargo project with an empty tokamak file the tokamak interface loaded after but was able to build a rust file but not run it because of unsaved buffer. Most of tokamak actually worked in that window so used it to create a new cargo binary from tokamak interface in atom. when it opened the new project in a new atom window the tokamak interface was gone and the plugin became unresponsive to any options again. It also seemed to die on the original atom window that was opened from terminal. I tried opening the project created by tokamak from terminal and tokamaks interface and plugin seems to be working but its a bit annoying to have it die when creating a new cargo project from tokamak. From what has worked so far it looks really good, thanks for contributing this and let me know if there is any input you need to try and get this issue resolved.

**Comment by detj** [comment]
I installed tokamak using `apm install tokamak` on OSX 10.11.6, but can't get it to work. The keyboard shortcuts mentioned in the site or of Windows and I tried every combination of option & command & shift but none of them seem to be working.
The README as absolutely no information on how to run commands. I don't see any command in the command palette in Atom when I search for tokamak. Usually, Atom packages list a few supported commands there. I mean the dialog accessed by `Command + Shift + P`

Is this just me? Did I miss something ?

**Comment by mrmcq2u** [comment]
@detj  - It's not just you, if you create an empty tokamak.toml file in the root directory of your cargo project and then open your .rs file from terminal tokamak might start up for that session? It has for me on fedora but when creating a new project within tokamak it will crash and need to be opened from terminal again.

**Comment by detj** [comment]
@mrmcq2u Any idea what the `tokamak.toml` file should contain. The larger question coming to my mind is why does tokamak needs a toml file? Its an Atom plugin, so shouldn't it have just commands to run for it to function. Maybe the commands are there, just not accessible.
Will have to dig into the source I guess to figure out.

**Comment by detj** [comment]
@mrmcq2u Okay, so I got a sample `tokamak.toml` file from the README and yeah, things started to move at least. I closed the Atom window and did an `atom .` from the dir containing `tokamak.toml` file and was presented with some errors which makes sense.

![screen shot 2016-08-18 at 1 15 51 pm](https://cloud.githubusercontent.com/assets/280236/17765925/292d582c-6546-11e6-8049-c949be045c63.png)

I guess I have to install a bunch of other stuff its complaining about. I don't have them installed.

**Comment by vertexclique** [comment]
Hello all;

TLDR; I am searching for contributors and maintainers to help with. Also want to implement a helper command line tool for Rust environment.

First for thanks using it to all. And thanks for all your feedbacks. I am the only one that is maintaining this and javascript is quite a b*tch for me...(for a low level guy). And you cannot do every tricks in javascript though IMHO. So I want to implement enhancements suggested in issues section as much as possible. This contextual hide functionality is suggested by a person in here => https://github.com/vertexclique/tokamak/issues/43 

So I want to continue this project as much as possible. Even I want to build a tool(helper) that helps Rust environment and projects. So if I build a command line tool, seems like we will solve most of the problems. If anyone wants to get involved for next leap I am greatly appreciate that person. Otherwise all the things are made as features are built in such a small time. There is no tests also which is **VERY VERY BAD**.

**Comment by locks** [comment]
What's the easiest way for someone to get involved?

**Comment by mrmcq2u** [comment]
@detj - try editing the settings for tokamak to look using fully qualified name e.g for rustup - /home/detj/.cargo/bin/rustup     , then when you have updated all of those open atom in root directory for cargo project again with atom . like you did before and things should more or less work hopefully.

**Comment by vertexclique** [comment]
@locks 
* Writing tests.
* Starting to write `plasma` project. (which will be the helper)
* Addressing the issues and fixing them with keeping `plasma will do the whole work in background` in the mind.

**Comment by flying-sheep** [comment]
so i had the same problem: without a tokamak.toml, nothing worked, and after creating one, it said “`undefined` has no property `save_buffers_before_run`”.

i think tokamak should also work without a `tokamak.toml` or with an empty one, or at least offer to fix the `tokamak.toml`.

**Comment by vertexclique** [comment]
Enabled global load by default in 0.3.2 version. I will document usage in wiki soon. Please update it for now.
